src/components/CandidateForm.jsx

```diff
@@ -26,14 +26,17 @@ const CandidateForm = ({ fields, onSubmit }) => {
   const onChange = (e) => {
     const newValues = [...values];
     const index = newValues.findIndex((value) => value.name === e.target.name);
-    newValues[index] = { ...newValues[index], value: e.target.value };
+    // newValues[index] = { ...newValues[index], value: e.target.value };
+    newValues.splice(index, 1, { ...newValues[index], value: e.target.value });
     setValues(newValues);
   };
 
   const onSelectChange = (selected) => {
     const newValues = [...values];
-    const index=newValues.findIndex(value => value.name === 'technologies')
-    newValues[index] = { name: 'technologies', value: selected}
+    const index = newValues.findIndex(value => value.name === 'technologies')
+    // newValues[index] = { name: 'technologies', value: selected}
+    newValues.splice(index, 1, { name: 'technologies', value: selected });
```

Comment:
I think in there you should map and format the values that come from the select. 

`value: selected.map((s) => s.value)`

I think this was the reason for the 500 error.